Add configure outbound proxy (#753)
AdguardTeam/KnowledgeBase: docs/adguard-for-linux/settings.md

- When running `adguard-cli config set listen_address <address>` where `<address>` is not `127.0.0.1`, AdGuard CLI will prompt for a username and password if proxy authentication is not already configured.
- When editing the config file directly, look for the `listen_auth`key. Set the `enabled` sub-key to `true`, and `username` and `password` to non-empty values.

## Configure outbound proxy

You can configure `outbound_proxy` if you want AdGuard CLI to work through another proxy server.

There are two ways to configure it:

### 1. Configure via URL (recommended)

Instead of setting each option step by step, you can set all parameters in a single line using a URL:

```sh
adguard-cli config set outbound_proxy https://user:pass@host:port
```

:::info

> Supported modes are HTTP, HTTPS, SOCKS4, SOCKS5.

Reviewer: ```suggestion
Supported modes are HTTP, HTTPS, SOCKS4, and SOCKS5.
```

:::

> You can also quickly enable, disable `outbound_proxy`:

Reviewer: так enable или disable?

Reviewer 2: `enable`, `disable` здесь переводимые - `включить` и `выключить`. Значение параметра - `true`/`false`

Reviewer: я не поняла. если вкл. или выкл, тогда нужно вместо запятой or

> You can also quickly enable, disable `outbound_proxy`:

Reviewer 2: ```suggestion
You can also quickly enable or disable `outbound_proxy`:
```

```sh
adguard-cli config set outbound_proxy false
```

Or quickly clear the settings:

```sh
adguard-cli config set outbound_proxy ""
```

### 2. Configure individual parameters

The ability to adjust specific parameters is also available:

```sh
adguard-cli config set outbound_proxy.enabled true
adguard-cli config set outbound_proxy.host localhost
adguard-cli config set outbound_proxy.port 3128
adguard-cli config set outbound_proxy.username user
adguard-cli config set outbound_proxy.password pass
```

Disable certificate verification for HTTPS proxies:

```sh
adguard-cli config set outbound_proxy.trust_any_certificate true
```

Enable SOCKS5 proxy for UDP traffic:

```sh
adguard-cli config set outbound_proxy.udp_through_socks5_enabled true
```

> :::info NOTE

Reviewer: ```suggestion
:::note
```

If your SOCKS5 proxy does not support UDP, connections may fail.

:::

### Checking the current configuration

To view the current outbound_proxy configuration, enter:

```sh
adguard-cli config show outbound_proxy
```

:::info Compatibility

> Configuring outbound_proxy via URL is available starting from version 1.1.26 nightly and stable release 1.1.

Reviewer: ```suggestion
Configuring `outbound_proxy` via URL is available starting from AdGuard for Linux v1.1.26 nightly and v1.1 stable release.
```
